### PR TITLE
Whizzml

### PR DESCRIPTION
**Pull request by mmerce**

### CODE REVIEW

**Comment by mmerce** [comment]
Tests pass in Python 2.7

File: docs/index.rst [hunk 1]
```diff
@@ -2029,6 +2035,163 @@ evaluation to use the remaining part of the dataset test data.
...
+
+Execute subcommand
+------------------
+
+This subcommand creates and executes scripts in WhizzML (BigML's automation
+language). With WhizzML you can program any specific workflow that involves
+Machine Learning resources like datasets, models, etc. You just write a
+script using the directives in the
+`reference manual <https://static.bigml.com/pdf/BigML_WhizzML_Reference.pdf>`_
+and upload it to BigML, where it will be available as more resource in
```
Comment (jaor) [inline docs/index.rst]: typo: one more

File: docs/index.rst [hunk 2]
```diff
@@ -2029,6 +2035,163 @@ evaluation to use the remaining part of the dataset test data.
...
+your dashboard. Scripts can also be shared and published in the gallery,
+so you can reuse other user's scripts and execute them. These operations
```
Comment (jaor) [inline docs/index.rst]: typo: other users' (or another user's)

File: docs/index.rst [hunk 3]
```diff
@@ -2029,6 +2035,163 @@ evaluation to use the remaining part of the dataset test data.
...
+can also be done using the `bigmler execute` subcommand.
+
+The simplest example is executing a basic code, like adding two numbers:
```
Comment (jaor) [inline docs/index.rst]: some basic code

File: docs/index.rst [hunk 4]
```diff
@@ -2029,6 +2035,163 @@ evaluation to use the remaining part of the dataset test data.
...
+
+Adding the ``--no-execute`` flag to the command will cause the process to
+stop right after the script creation. You can also compile your code as a
+library to be used in many scripts by setting the ``--to-library`` flag.
+
+.. code-block:: bash
+
+    bigmler execute --code-file my_library.whizzml --no-execute --to-library
+
+Existing scripts can be referenced for execution with the ``--script`` option
```
Comment (jaor) [inline docs/index.rst]: just double checking here, but i thought `--to-library` would imply `--no-execute` (libraries cannot be executed)

File: docs/index.rst [hunk 5]
```diff
@@ -2029,6 +2035,163 @@ evaluation to use the remaining part of the dataset test data.
...
+
+    bigmler execute --script script/50a2bb64035d0706db000643
+
+or the script ID can be read from a file:
+
+.. code-block:: bash
+
+    bigmler execute --scripts simple_exe/scripts
+
+However, the script we used as example is too simple. In general, scripts
```
Comment (jaor) [inline docs/index.rst]: I would write this as: "The script we used as an example is very simple and needs no additional parameter.  But, in general, scripts.."

(whatever you like better is fine for me)

File: docs/index.rst [hunk 6]
```diff
@@ -2029,6 +2035,163 @@ evaluation to use the remaining part of the dataset test data.
...
+      "description": "second number to add"}]
+
+and ``my_outputs_dec.json``
+
+.. code-block:: json
+
+    [["addition"]]
+
+so that the value of the ``addition`` variable would be returned as
+output in the execution results.
```
Comment (jaor) [inline docs/index.rst]: hmmm, i think this is not right.  when creating a script or library, the outputs are described using a json similar to that of the inputs (without "default" and optional "type")... when creating an execution one can provide an alternative list of outputs, that can be either simple names (as in your example), or pairs ["name", "type"], although we never check the type.

File: docs/index.rst [hunk 7]
```diff
@@ -2029,6 +2035,163 @@ evaluation to use the remaining part of the dataset test data.
...
+        "source": {
+        "project": "project/54d9553bf0a5ea5fc0000016"
+        },
+        "ensemble": {
+        "number_of_models": 100, "sample_rate": 0.9
+        }
+    }
+
+the source created by the script will be associated to the given project
+and the ensemble will have 100 models and a 0.9 sample rate.
```
Comment (jaor) [inline docs/index.rst]: perhaps i would add here "(unless the source code in your script explicitly specifies a different value, in which case it takes precedence over these defaults)."  but maybe it's already clear from the discussion above...

File: docs/index.rst (hunk 4 above)
Comment (mmerce) [inline docs/index.rst]: It does

**Pull request merged**